src/build/env.py

```diff
@@ -38,6 +42,24 @@ def _restore_env(self):  # type: () -> None
     def _get_env_path(self, path):  # type: (str) -> Optional[str]
         return sysconfig.get_path(path, vars=self._env_vars)
 
+    def _symlink_relative(self, path):  # type: (Optional[str]) -> None
+        if not path:  # pragma: no cover
+            return
+
+        prefix = sysconfig.get_config_var('prefix')
+        if prefix and path and path.startswith(prefix):
+            new_path = os.path.join(self._path, path[len(prefix + os.pathsep):])
+            if not os.path.exists(new_path):
+                try:
+                    os.makedirs(os.path.dirname(new_path))
+                except FileExistsError:
+                    pass
+                if os.name == 'nt':
+                    import shutil
+                    shutil.copytree(path, new_path)
```

Comment:
Windows does support symlinks for a while now https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/ 🤔 This is kinda presemptious it does not. Here's what virtualenv does https://github.com/pypa/virtualenv/blob/master/src/virtualenv/info.py#L32-L51 to check for this.

Reply:
IIRC the github actions failed when using symlinks.

> This new symlinks support first shipped in Windows 10 Insiders Build 14972, and will be formally delivered in Windows 10 Creators Update.

For older versions, they need to be manually enabled. I want to make sure it works for everyone, so I went with the Windows approach of just copying everything everywhere.

Reply:
We can add support for checking if symlinks are available, but I'd prefer to do it in another PR as this is already huge as is. And that would only be an optimization.

Reply:
If it's only an optimization drop it for Linux too, always copy. There's no reason to start discrimnating on behaviour between OSs.

Reply:
> For older versions, they need to be manually enabled. I want to make sure it works for everyone, so I went with the Windows approach of just copying everything everywhere.

What older versions? Older Windows? That's the point of my test, I actively check if it works or not. And use copy if symlink does not work.

Reply:
In Linux we can safely assume it works, Windows is problematic. The check is pretty small and reasonable, I think. It is already an optimization, it is not perfect but gives us good result for a small complexity hit. I can remove the behavior but that feels like moving backwards to me, let me know if you want me to do it.

> What older versions? Older Windows? That's the point of my test, I actively check if it works or not. And use copy if symlink does not work.

Yes, older Windows. I am not saying we shouldn't add a test like yours, we totally should, I am just saying that this PR is already big as is and is severely blocking future progress. As this is just a small optimization with small impact on the rest of the project, I just suggested doing it later, once we sort out the more pressing issues. What is your position on this? In your opinion, does this make sense?

Reply:
> I can remove the behavior but that feels like moving backwards to me, let me know if you want me to do it.

I'd prefer to just pull in the code as is from virtualenv and do the prefer symlink use copy.

> I am just saying that this PR is already big as is and is severely blocking future progress.

Part of the reason it's blocking progress is because we're arguing about the scope. Rather than to do this, let's just fix the raised issues and merge it, and not have follow-up issues.